userspace/libsinsp/tuples.h

```diff
@@ -48,18 +49,31 @@ typedef struct ipv4net
 	uint32_t m_netmask; ///< Subnet mask
 }ipv4net;
 
-typedef struct _ipv6addr
+struct ipv6addr
 {
```

Comment:
In the earlier version of this PR (https://github.com/falcosecurity/libs/pull/210/files), I mentioned that the size of this struct must always be exactly 16 bytes. Otherwise, the sizeof(ipv6addr) will be incorrect at https://github.com/falcosecurity/libs/pull/233/files#diff-a5a0f5b0d3df88e4d2ad24728c1004e1db648bdade8dba9268ad981696172f3bR165.

You mentioned that you still wanted to use sizeof() and assume that this struct would just be plain old data. Can you add a comment here that warns future devs that this struct *must* only contain the 16 bytes for the ipv6 address and nothing else, or that the sizeof() in value_parser.cpp must change?

Reply:
ok, I am adding compile time check but I believe that what does matter is the object memory layout not the size. in this case must be plain - all members are scalars and no virtual inheritance - Plain Old Data

Reply:
Right, exactly. I see the changes, thank you!